File: examples/python/youtube.py
import requests
import os
import base64

API_KEY = os.getenv('COGNIMA_API_KEY', 'ck_your_api_key')
BASE_URL = 'https://cog2.cognima.com.br/api/v1'

def search_youtube():
    url = f'{BASE_URL}/youtube/search'
    headers = {
        'X-API-Key': API_KEY,
        'Content-Type': 'application/json'
    }
    data = {
        'query': 'python programming tutorial'
    }
    
    response = requests.post(url, json=data, headers=headers)
    
    if response.status_code == 200:
        result = response.json()
        print(f"Found {result['data']['count']} results\n")
        for i, video in enumerate(result['data']['results'][:5], 1):
            print(f"{i}. {video['title']}")
            print(f"   Channel: {video['channel']['name']}")
            print(f"   Views: {video['views']:,}")
            print(f"   URL: {video['url']}\n")
    else:
        print(f"Error {response.status_code}: {response.json()}")

def download_mp3():
    url = f'{BASE_URL}/youtube/mp3'
    headers = {
        'X-API-Key': API_KEY,
        'Content-Type': 'application/json'
    }
    data = {
        'url': 'https://www.youtube.com/watch?v=dQw4w9WgXcQ'
    }
    
    response = requests.post(url, json=data, headers=headers)
    
    if response.status_code == 200:
        result = response.json()
        print('Audio downloaded!')
        print(f"Title: {result['data']['title']}")
        print(f"Duration: {result['data']['duration']}s")
        print(f"Filename: {result['data']['filename']}")
        
        # Save file
        audio_data = base64.b64decode(result['data']['buffer'])
        with open(result['data']['filename'], 'wb') as f:
            f.write(audio_data)
        print(f"Saved to {result['data']['filename']}")
    else:
        print(f"Error {response.status_code}: {response.json()}")

def download_mp4():
    url = f'{BASE_URL}/youtube/mp4'
    headers = {
        'X-API-Key': API_KEY,
        'Content-Type': 'application/json'
    }
    data = {
        'url': 'https://www.youtube.com/watch?v=dQw4w9WgXcQ',
        'quality': '720p'
    }
    
    response = requests.post(url, json=data, headers=headers)
    
    if response.status_code == 200:
        result = response.json()
        print('Video downloaded!')
        print(f"Title: {result['data']['title']}")
        print(f"Quality: {result['data']['quality']}")
    else:
        print(f"Error {response.status_code}: {response.json()}")

if __name__ == '__main__':
    search_youtube()

File: examples/nodejs/youtube.js
const axios = require('axios');
const fs = require('fs');

const API_KEY = process.env.COGNIMA_API_KEY || 'ck_your_api_key';
const BASE_URL = 'https://cog.api.br/api/v1';

async function searchYouTube() {
  try {
    const response = await axios.post(
      `${BASE_URL}/youtube/search`,
      { query: 'javascript tutorial' },
      {
        headers: {
          'X-API-Key': API_KEY,
          'Content-Type': 'application/json'
        }
      }
    );
    
    console.log(`Found ${response.data.data.count} results\n`);
    response.data.data.results.slice(0, 5).forEach((video, i) => {
      console.log(`${i + 1}. ${video.title}`);
      console.log(`   Channel: ${video.channel.name}`);
      console.log(`   Views: ${video.views.toLocaleString()}`);
      console.log(`   URL: ${video.url}\n`);
    });
  } catch (error) {
    console.error('Error:', error.response?.data || error.message);
  }
}

async function downloadMp3() {
  try {
    const response = await axios.post(
      `${BASE_URL}/youtube/mp3`,
      { url: 'https://www.youtube.com/watch?v=dQw4w9WgXcQ' },
      {
        headers: {
          'X-API-Key': API_KEY,
          'Content-Type': 'application/json'
        }
      }
    );
    
    const data = response.data.data;
    console.log('Audio downloaded!');
    console.log(`Title: ${data.title}`);
    console.log(`Duration: ${data.duration}s`);
    console.log(`Filename: ${data.filename}`);
    
    // Save file
    const audioBuffer = Buffer.from(data.buffer, 'base64');
    fs.writeFileSync(data.filename, audioBuffer);
    console.log(`Saved to ${data.filename}`);
  } catch (error) {
    console.error('Error:', error.response?.data || error.message);
  }
}

async function downloadMp4() {
  try {
    const response = await axios.post(
      `${BASE_URL}/youtube/mp4`,
      { 
        url: 'https://www.youtube.com/watch?v=dQw4w9WgXcQ',
        quality: '720p'
      },
      {
        headers: {
          'X-API-Key': API_KEY,
          'Content-Type': 'application/json'
        }
      }
    );
    
    console.log('Video downloaded!');
    console.log(`Title: ${response.data.data.title}`);
    console.log(`Quality: ${response.data.data.quality}`);
  } catch (error) {
    console.error('Error:', error.response?.data || error.message);
  }
}

searchYouTube();
